php-frontend/src/main/java/org/sonar/php/parser/PHPLexicalGrammar.java

```diff
@@ -0,0 +1,347 @@
+  TRY_STATEMENT,
+  CATCH_BLOCK,
+  FINALLY_STATEMENT,
+  THROW_STATEMENT,
+  GOTO_STATEMENT,
+  YIELD_STATEMENT,
+  GLOBAL_STATEMENT,
+  GLOBAL_VAR,
+  STATIC_STATEMENT,
+  STATIC_VAR,
+  UNSET_VARIABLE_STATEMENT,
+  UNSET_VARIABLES,
+  PARAMETER_LIST,
+  PARAMETER,
+  OPTIONAL_CLASS_TYPE,
+  METHOD_DECLARATION,
+  METHOD_BODY,
+  VARIABLE_MODIFIERS,
+  CLASS_VARIABLE_DECLARATION,
+  VARIABLE_DECLARATION,
+  MEMBER_MODIFIER,
+  CLASS_CONSTANT_DECLARATION,
+  MEMBER_CONST_DECLARATION,
+  TRAIT_USE_STATEMENT,
+  TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED,
+  TRAIT_METHOD_REFERENCE,
+  TRAIT_ALIAS,
+  TRAIT_PRECEDENCE,
+  TRAIT_ADAPTATION_STATEMENT,
+  TRAIT_ADAPTATIONS,
+  INLINE_HTML_STATEMENT,
+
+  /**
+   * Expression
+   */
+  PRIMARY_EXPRESSION,
+  MEMBER_EXPRESSION,
+  ALIAS_VARIABLE,
+  VARIABLE_NAME,
+  COMPUTED_VARIABLE_NAME,
+  VARIABLE_WITHOUT_OBJECTS,
+  EXPRESSION_STATEMENT,
+  CLASS_MEMBER_ACCESS,
+  OBJECT_MEMBER_ACCESS,
+  SIMPLE_INDIRECT_REFERENCE,
+  STATIC_MEMBER,
+  COMPOUND_VARIABLE,
+  CLASS_NAME,
+  FUNCTION_CALL_PARAMETER_LIST,
+  PARAMETER_LIST_FOR_CALL,
+  DIMENSIONAL_OFFSET,
+  STATIC_SCALAR,
+  OBJECT_DIM_LIST,
+  PARENTHESIS_EXPRESSION,
+  COMBINED_SCALAR,
+  COMMON_SCALAR,
+  BOOLEAN_LITERAL,
+  LEXICAL_VARS,
+  LEXICAL_VAR_LIST,
+  LEXICAL_VAR,
+  LOGICAL_XOR_EXPR,
+  LOGICAL_OR_EXPR,
+  LOGICAL_OR_OPERATOR,
+  BITEWISE_AND_EXPR,
+  BITEWISE_XOR_EXPR,
+  BITEWISE_OR_EXPR,
+  LOGICAL_AND_EXPR,
+  LOGICAL_AND_OPERATOR,
+  CONDITIONAL_EXPR,
+  ASSIGNMENT_EXPR,
+  MULTIPLICATIVE_EXPR,
+  MULIPLICATIVE_OPERATOR,
+  ADDITIVE_EXPR,
+  ADDITIVE_OPERATOR,
+  SHIFT_EXPR,
+  SHIFT_OPERATOR,
+  RELATIONAL_EXPR,
+  RELATIONAL_OPERATOR,
+  EQUALITY_EXPR,
+  EQUALITY_OPERATOR,
+  CONCATENATION_EXPR,
+  POSTFIX_EXPR,
+  UNARY_EXPR,
+  ASSIGNMENT_BY_REFERENCE,
+  ASSIGNMENT_OPERATOR,
+  COMPOUND_ASSIGNMENT,
+  CAST_TYPE,
+  LOGICAL_ASSIGNMENT,
+  INTERNAL_FUNCTION,
+  NEW_EXPR,
+  COMBINED_SCALAR_OFFSET,
+  ARRAY_PAIR_LIST,
+  ARRAY_PAIR,
+  EXIT_EXPR,
+  LIST_EXPR,
+  LIST_ASSIGNMENT_EXPR,
+  ASSIGNMENT_LIST_ELEMENT,
+  ASSIGNMENT_LIST,
+  FUNCTION_EXPRESSION,
+  COMPLEX_ENCAPS_VARIABLE,
+  SEMI_COMPLEX_ENCAPS_VARIABLE,
+  SEMI_COMPLEX_RECOVERY_EXPRESSION,
+  SIMPLE_ENCAPS_VARIABLE,
+  ENCAPS_LIST,
+  ENCAPS_VAR_OFFSET,
+  ENCAPS_VAR,
+  ENCAPS_DIMENSIONAL_OFFSET,
+  ENCAPS_OBJECT_MEMBER_ACCESS,
+
+  NULL,
+  CLASS_CONSTANT,
+  FILE_CONSTANT,
+  DIR_CONSTANT,
+  FUNCTION_CONSTANT,
+  LINE_CONSTANT,
+  METHOD_CONSTANT,
+  NAMESPACE_CONSTANT,
+  TRAIT_CONSTANT,
+  NEXT_IS_DOLLAR,
+  DOUBLE_QUOTE,
+  REGULAR_STRING_LITERAL,
+  VARIABLE_VARIABLE_DOLLAR,
+
+  WHITESPACES,
+  EXPRESSION;
+
+
+
+  public static LexerlessGrammar createGrammar() {
+    return createGrammarBuilder().build();
+  }
+
+  public static LexerlessGrammarBuilder createGrammarBuilder() {
+    LexerlessGrammarBuilder b = LexerlessGrammarBuilder.create();
+
+    b.rule(COMPILATION_UNIT).is(b.optional(SCRIPT), SPACING, EOF);
+    // FIXME should be TOP_STATEMENT_LIST
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)